src/game/800.c

```diff
@@ -1,34 +1,104 @@
 #include "game.h"
 
-INCLUDE_ASM("asm/game/nonmatchings/800", func_80021CA4);
+s32 func_80021CA4(void) {
+    func_80099AFC();
+    func_80059CCC();
+    *(s32*)0x1F8000A0 = 0xF00DFACE;
+    *(s32*)0x1F8000B8 = 0xF00DFACE;
+    func_80059E58();
+    return 0;
+}
 
 INCLUDE_ASM("asm/game/nonmatchings/800", func_80021CEC);
 
 INCLUDE_ASM("asm/game/nonmatchings/800", func_80022AE8);
 
-INCLUDE_ASM("asm/game/nonmatchings/800", func_80022B5C);
+void func_80022B5C(void) {
+    u16 var_v1;
+
+    if (D_800EEE64 == 3) {
+        var_v1 = D_800EEE5E;
+    } else {
+        var_v1 = 0;
+    }
+    func_800234C4(&D_800EE020, *(var_v1 + &D_800CDF5C), 0x10);
```

Comment:
```suggestion
func_800234C4(&D_800EE020, D_800CDF5C[var_v1], 0x10);
```
Remember changing the type from s32 D_800CDF5C to s32 D_800CDF5C[]